### PR TITLE
Add custom approval rule by app

### PR DESCRIPTION
**Pull request by quynhnguyen68**
In this MR, I added the instruction:
-  How to add a custom condition in Approval Rule of B2B Order Approval by Shopware App
- Adjust the instruction of how to add custom condition Approval Rule of B2B Order Approval by plugin since class `OrderApprovalRule` is removed.

### CODE REVIEW

**Review by bojanrajh — APPROVED** [review]

File: products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md [hunk 1]
```diff
@@ -70,14 +74,216 @@ class CartAmountRule extends OrderApprovalRule
...
+::: info
+Note that the value field is only supported with number fields (`float`, `int`), text fields (`text`), single-selection fields (`single-select`), and multi-selection fields (`multi-select`).
+:::
+
+This is how the custom condition appears on the Approval Rule detail page, the `Total cart amount` condition in added to condition selection list. In this case, the value is displayed as a number field that allows decimals, with each increment being 0.01. This is because the `float` tag is defined in the manifest file.
+
+![App Approval Rule Condition](../../../../../assets/approval-rule-condition.png)
+
+#### Scripts
+Script logic is similar to [Add custom rule conditions](../../../../../guides/plugins/apps/rule-builder/add-custom-rule-conditions.md). Let's continue with our example by creating a script file that compares the current shopping cart's total price.
```
[the remark this answers is not in the document]
Comment (sushmangupta) [reply products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md:136]: script file that compares the current shopping cart's total price against what ?

**Review by sushmangupta — CHANGES_REQUESTED** [review]

File: products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md [hunk 2]
```diff
@@ -70,14 +74,216 @@ class CartAmountRule extends OrderApprovalRule
...
+
+Similar to [Add custom rule conditions](../../../../../guides/plugins/apps/rule-builder/add-custom-rule-conditions.md), a condition's manifest file has `identifier`, `name` and `script`. 
+
+The `name` is a descriptive and translatable name for the condition. The name will be shown within the Condition's selection in the Storefront Approval Rule detail or create page.
+
+Within the script tag, the path of the twig script file must include the `/approval-rule-conditions/` prefix.
+
+Let's create a custom condition that compares the total price of the current shopping cart.
+
+<<< @/docs/snippets/config/app/approval-rule-conditions.xml
```
[the remark this answers is not in the document]
Comment (quynhnguyen68) [reply products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md:121]: I use it to show sample manifest file

Screenshot 2024-06-15 at 17 10 39

File: products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (quynhnguyen68) [reply products/extensions/b2b-components/order-approval/guides/04-add-new-approval-condition.md]: @sushmangupta I already edit this sentence 
"Let's continue with our example by creating a script file that compares the current shopping cart's total price with the pre-established value in the approval rule."

**Review by sushmangupta — APPROVED** [review]
Thanks for updating the docs!

**Pull request merged**